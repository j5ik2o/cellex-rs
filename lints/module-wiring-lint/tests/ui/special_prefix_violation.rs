#![feature(register_tool)]
#![register_tool(module_wiring)]
#![warn(module_wiring::no_parent_reexport)]

mod child {
  pub struct Thing;
}

pub use self::child::Thing;

fn main() {}
