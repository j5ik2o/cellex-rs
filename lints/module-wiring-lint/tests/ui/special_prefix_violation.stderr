warning: 再エクスポートは末端モジュールの直属親以外では禁止です
  --> $DIR/special_prefix_violation.rs:9:1
   |
LL | pub use self::child::Thing;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: ルール: 原則ルール: 再エクスポート禁止
   = note: 詳細: 特殊パス（self / super / crate）を経由した再エクスポートは禁止されています
   = help: 末端モジュールの直属親からのみ再エクスポートしてください
   = note: `#[warn(no_parent_reexport)]` on by default

warning: 1 warning emitted

